seismicpro/refractor_velocity/metrics.py

```diff
@@ -0,0 +1,494 @@
+            Array indicating whether each trace in the gather represents an outlier.
+        """
+        rv_times = refractor_velocity(gather["offset"])
+        gather_times = gather[first_breaks_col]
+        correct_uphole = (correct_uphole if correct_uphole is not None
+                          else ("SourceUpholeTime" in gather.available_headers
+                                and refractor_velocity.is_uphole_corrected))
+        if correct_uphole:
+            gather_times = gather_times + gather["SourceUpholeTime"]
+        return self._calc(rv_times, gather_times, self.threshold_times)
+
+    def plot_gather(self, *args, **kwargs):
+        """Plot the gather with highlighted outliers on top of the gather plot."""
+        super().plot_gather(*args, top_header=False, **kwargs)
+
+    def plot_refractor_velocity(self, coords, ax, index, **kwargs):
+        """Plot the refractor velocity curve and show the threshold area used for metric calculation."""
+        threshold_times = kwargs.get("threshold_times", self.threshold_times)
+        return super().plot_refractor_velocity(
+            coords, ax, index, threshold_times=threshold_times, **kwargs)
+
+
+class FirstBreaksAmplitudes(RefractorVelocityMetric):
+    """Mean amplitude of the signal in the moment of first break after maxabs scaling."""
+
+    name = "first_breaks_amplitudes"
+    vmin = 0
+    vmax = 0.5
+    is_lower_better = None
+
+    @staticmethod
+    @njit(nogil=True)
+    def _calc(gather_data, gather_times, sample_rate):
+        """Calculate signal amplitudes at first break times."""
+        gather_data = scale_maxabs(gather_data, min_value=None, max_value=None, q_min=0, q_max=1,
+                                   clip=False, tracewise=True, eps=1e-10)
+        ix = gather_times / sample_rate
+        if np.any(gather_data.shape[1] < ix) or np.any(ix < 0):
+            raise IndexError("First breaks are out of bounds")
+        res = np.empty_like(ix, dtype=gather_data.dtype)
+        for i, idx in enumerate(ix):
+            prev_idx, next_idx = floor(idx), ceil(idx)
+            weight = next_idx - idx
+            res[i] = gather_data[i, prev_idx] * weight + gather_data[i, next_idx] * (1 - weight)
+        return res
+
+    def calc(self, gather, refractor_velocity, first_breaks_col, correct_uphole):
+        """Return signal amplitudes at first break times.
+
+        Parameters
+        ----------
+        gather : Gather
+            A seismic gather to get offsets and times of first breaks from.
+        first_breaks_col : str, optional, defaults to :const:`~const.HDR_FIRST_BREAK`
+            Column name from `survey.headers` where times of first break are stored.
+
+        Returns
+        -------
+        metric : np.ndarray of float
+            Signal amplitudes for each trace in the gather.
+        """
+        _ = refractor_velocity, correct_uphole
+        res = self._calc(gather.data.copy(), gather[first_breaks_col], gather.sample_rate)
```

Comment:
```suggestion
        res = self._calc(gather.data, gather[first_breaks_col], gather.sample_interval)
```
And let's move `copy` inside the method